src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/SetAzureRmServiceFabricUpgradeType.cs

```diff
@@ -0,0 +1,69 @@
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ServiceFabric.Common;
+using Microsoft.Azure.Commands.ServiceFabric.Models;
+using Microsoft.Azure.Management.ServiceFabric.Models;
+
+namespace Microsoft.Azure.Commands.ServiceFabric.Commands
+{
+    [Cmdlet(VerbsCommon.Set, CmdletNoun.AzureRmServiceFabricUpgradeType), OutputType(typeof(PSCluster))]
+    public class SetAzureRmServiceFabricUpgradeType : ServiceFabricClusterCmdlet
+    {
+        private const string AutomaticSet = "Automatic";
+        private const string ManualSet = "Manual";
+
+        [Parameter(Mandatory = true, Position = 0, ParameterSetName = AutomaticSet, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Specify the name of the resource group.")]
+        [Parameter(Mandatory = true, Position = 0, ParameterSetName = ManualSet, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Specify the name of the resource group.")]
+        [ValidateNotNullOrEmpty()]
+        public override string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, Position = 1, ParameterSetName = AutomaticSet, ValueFromPipelineByPropertyName = true,
+                   HelpMessage = "Specify the name of the cluster")]
+        [Parameter(Mandatory = true, Position = 1, ParameterSetName = ManualSet, ValueFromPipelineByPropertyName = true,
+                   HelpMessage = "Specify the name of the cluster")]
+        [ValidateNotNullOrEmpty()]
+        public override string ClusterName { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = AutomaticSet,
+                   HelpMessage = "ClusterUpgradeMode")]
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = ManualSet,
+                   HelpMessage = "Cluster upgrade mode, e.g. Automatic or Manual")]
+        [ValidateNotNullOrEmpty()]
+        public ClusterUpgradeMode UpgradeMode { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = ManualSet,
+                   HelpMessage = "Cluster code version")]
+        [ValidateNotNullOrEmpty()]
+        [Alias("ClusterCodeVersion")]
+        public string Version { get; set; }
```

Comment:
the version changing while we release our products, so i don't think we should a validate set